index.js

```
@@ -116,82 +131,119 @@ app.use(Helmet())
+      walletRequest.address = address
+    } catch (e) {
+      return reject('Invalid address supplied')
+    }
+
+    /* Verify that the caller supplied us with an acceptable callback
      URL that we'll post back to later */
-  if (callback) {
-    if (callback.substring(0, 4).toLowerCase() !== 'http') {
-      logHTTPError(req, 'Invalid callback URL supplied')
-      return res.status(400).send()
+    if (callback) {
+      if (callback.substring(0, 4).toLowerCase() !== 'http') {
+        return reject('Invalid callback URL supplied')
+      }
     }
-  }
 
-  /* Verify that the caller has supplied a valid amount to request */
-  if (!atomicAmount || atomicAmount === 0 || atomicAmount < 0) {
-    logHTTPError(req, 'Invalid amount requested')
-    return res.status(400).send()
-  }
+    walletRequest.callback = callback
+
+    /* Verify that the caller has supplied a valid amount to request */
+    if (!atomicAmount || atomicAmount === 0 || atomicAmount < 0) {
+      return reject('Invalid amount requested')
+    }
+
+    walletRequest.amount = atomicAmount
 
-  /* If the caller has supplied the number of confirmations that they
+    /* If the caller has supplied the number of confirmations that they
      are willing to wait and override our defaults, then we're going
      to validate that it's okay. */
-  var requestConfirmations
-  if (confirmations !== false) {
-    /* If the caller requested 0 or less or more confirmations than we
+    var requestConfirmations
+    if (confirmations !== false) {
+      /* If the caller requested 0 or less or more confirmations than we
        allow, we're going to reject their request */
-    if (confirmations < 0 || confirmations > Config.maximumConfirmations) {
-      logHTTPError(req, 'Invalid confirmations requested')
-      return res.status(400).send()
+      if (confirmations < 0 || confirmations > Config.maximumConfirmations) {
+        return reject('Invalid confirmations requested')
```

reject should return an error. ie.

`reject(new Error('Invalid address supplied'))`